Functions/Mock.ps1

```diff
@@ -226,6 +226,29 @@ function Create-MockHook ($contextInfo, $InvokeMockCallback) {
     }
 "@
 
+    $functionName = [Guid]::NewGuid().Guid
+    $code = @"
```

Comment:
This is the essence of the change. By introducing a new level of indirection on the function definition we are able to leverage the `$args` values unfiltered to determine the expected dynamic parameters